sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveCatalog.scala

```diff
@@ -85,6 +85,13 @@ private[spark] class HiveCatalog(client: HiveClient) extends ExternalCatalog wit
     withClient { getTable(db, table) }
   }
 
+  /**
+   * Switch our client to one that belongs to the new session.
+   */
+  def newSession(newClient: HiveClient): this.type = {
```

Comment:
yea - if it is going to take a while feel free to do as an add-on pr. it'd be fix tests and great to merge this one asap.